docs/designers-developers/developers/block-api/block-deprecation.md

```diff
@@ -7,11 +7,11 @@ When updating static blocks markup and attributes, block authors need to conside
+Deprecations do not operate as a chain of updates in the way other software data updates, like database migrations, do. At first glance, it is easy to think that each deprecation is going to make the required changes to the data and then hand this new form of the block onto the next deprecation to make its changes. What happens is that each deprecation is passed the original saved content, and if its `save` method produces a valid block it is used to parse the block attributes. If it has a `migrate` method it will also be run, and the data is then passed back to the current block `save` function to create the new valid version of the block.
 
-It is also important to note that if a deprecation's `save` method does not produce a valid block then it is skipped, including its `migration` method, even if `isEligible` returns true. This means that if you have a number of deprecations for a block and want to perform a new migration, like moving content to InnerBlocks, you may need to include the `migrate` method in multiple deprecations in order for it to be applied to all previous version of the block.
+It is also important to note that if a deprecation's `save` method does not produce a valid block then it is skipped, including its `migrate` method, even if `isEligible` returns true. This means that if you have several deprecations for a block and want to perform a new migration, like moving content to InnerBlocks, you may need to include the `migrate` method in multiple deprecations for it to be applied to all previous versions of the block.
 
-For ease of management it is recommended that you version the deprecations by saving each deprecation into a separate file with the name of the block version it applies to. Then import each of these objects into the `deprecated` property array mentioned below. It is also recommeded to keep [fixtures](https://github.com/WordPress/gutenberg/tree/master/packages/e2e-tests/fixtures/blocks) which contain the different versions of the block content to allow you to easily test that new deprecations and migrations are working across all previous versions of the block.
+For ease of management, it is recommended that you version the deprecations by saving each deprecation into a separate file with the name of the block version it applies to. Then import each of these objects into the `deprecated` property array mentioned below. It is also recommended to keep [fixtures](https://github.com/WordPress/gutenberg/tree/master/packages/e2e-tests/fixtures/blocks) which contain the different versions of the block content to allow you to easily test that new deprecations and migrations are working across all previous versions of the block.
```

Comment:
A brief example might be good, as I think it's quite hard to visualize what a 'block version' is. Some might misinterpret this as an explicit version number in the deprecation object, but I think what you're getting at is better semantics using variable names.

I guess this is the only bit of the PR that might need more discussion, as it's not current practice.

I'd be interested in other thoughts. Something I considered once before was to use comments like:
```js
// Initial version of block
// ...
// 1. Deprecation after captions added.
```

I like the idea of numbering matching the fixtures. Almost makes me think the fixture html should live alongside the deprecation. An advantage of linking them together in some way is that the tests could then be generic deprecation tests that even work on a third party block that's defined the same way.

Could also be nice to link to the readme in the fixtures folder, as they're quite difficult to understand for a newcomer.

Reply:
thanks @talldan, that is a good idea to expand this with an example ... we went with this approach in the gallery refactor - https://github.com/WordPress/gutenberg/pull/25940/files#diff-04bda7ff65177c115240f624f7b131cb24350a290b83d9f16f0462bc8a111132R4  - what do you think about this as a suggested way of handling it?

Reply:
It makes sense to have some way to indicate which deprecation came first. For core blocks I think I've noticed usually the earliest deprecation is last in the array, but as you say, that's not a clear convention for contributors.

I'm not overly opinionated on the method used to improve this. I anticipate some might not like the idea of splitting the deprecations into multiple files, but I personally am ok with that. Having said that, the same system could easily be used within one file. e.g.:

```js
const v1 = { // ... };
const v2 = { // ... };

const deprecated = [ v1, v2 ];
export default deprecated;
```

Worth pinging others on this like @gziolo who's done loads of good work on improving the block structure and bringing consistency across blocks.

Reply:
Given this feedback I think that keeping in a single file, but assigning each to a const may be an easier step than putting each in its own file, so have reworded and provided and example for this ... and have also refactored the gallery PR to use a single file instead of multiple.